js/ui/grid_core/ui.grid_core.column_chooser.js

```diff
@@ -370,24 +359,46 @@ const columnChooserMembers = {
         };
     },
 
+    _updateItems: function() {
+        const isSelectMode = this.isSelectMode();
+        const chooserColumns = this._columnsController.getChooserColumns(isSelectMode);
+        const items = processItems(this, chooserColumns);
+
+        this._columnChooserList.option('items', items);
+    },
+
+    _updateItemSelection: function(columnIndex) {
+        const isRecursive = this.option('columnChooser.selection.recursive');
+        const column = this._columnsController.columnOption(columnIndex);
+
+        const isRecursiveWithColumns = isRecursive && column.hasColumns;
+
+        if(!isRecursiveWithColumns) {
+            column.visible ?
+                this._columnChooserList.selectItem(columnIndex) :
+                this._columnChooserList.unselectItem(columnIndex);
```

Comment:
Minor comment.

Let's change a little formatting here:
```suggestion
            column.visible
                ? this._columnChooserList.selectItem(columnIndex)
                : this._columnChooserList.unselectItem(columnIndex);
```